code/obj/machinery/manufacturer.dm

```diff
@@ -407,8 +407,9 @@ TYPEINFO(/obj/machinery/manufacturer)
 	proc/blueprints_as_list	(var/list/L, mob/user, var/static_elements = FALSE)
 		var/list/as_list = list()
 		for (var/datum/manufacture/M as anything in L)
-			if (isnull(M.category)) // fix for not displaying blueprints/manudrives
+			if (!(M.category in src.categories)) // fix for not displaying blueprints/manudrives
 				M.category = "Miscellaneous"
+				// "WARN: bad category on TYPEPATH"
```

Comment:
```suggestion
```
No need for commented out debug code here.

Reply:
Oh yes i meant to ask mordent, is there a way to have a warning here? In case someone tries to define a new category that doesnt exist yet